### PR TITLE
Fix git secrets and git hooks

### PR DESCRIPTION
**Pull request by dmohs**
Git hooks installed by the `shared-git-hooks` NPM module are done within the docker container, so the symlinks are not correct on the host machine. For me, that made them fail silently. Additionally, if I don't start the UI, they don't get installed.

Now they are installed using a local git config (available since version 2.9). I also fixed some other issues and factored out some common code.

When this PR is merged, each developer will want to run
```bash
find .git/hooks -type l -exec rm {} \;
```
to clean out the old hooks.

### CODE REVIEW

File: hooks/apply-git-secrets.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (markfickett) [inline hooks/apply-git-secrets.sh]: Yay!

File: api/libproject/devstart.rb [hunk 2]
```diff
@@ -3,6 +3,13 @@
 require "optparse"
 require "tempfile"
 
+def ensure_git_hooks()
```
Comment (markfickett) [inline api/libproject/devstart.rb]: Looks like we'll need a shared place for Ruby scripts a level up?

File: hooks/pre-commit [hunk 3]
```diff
@@ -1,20 +1,8 @@
...
-} || {
-  :
-}
-git secrets --add 'private_key'
-git secrets --add 'private_key_id'
-git secrets --add --allowed --literal "git secrets --add 'private_key'"
-git secrets --add --allowed --literal "git secrets --add 'private_key_id'"
+./hooks/apply-git-secrets.sh
 
 git secrets --pre_commit_hook -- "$@"
```
Comment (dmohs) [inline hooks/pre-commit]: The additional code running below this line caused the commit to succeed even though git-secrets complained. That didn't seem like the right behavior. Hence the `set -euo pipefail` above.

**Comment by dmohs** [comment]
@blrubenstein Does my last commit address the issue with devstart.rb?

File: api/libproject/devstart.rb (hunk 2 above)
Comment (dmohs) [reply api/libproject/devstart.rb]: Can we punt on this? My intuition is that there won't be much code sharing, but I would hate to have any duplication that is non-trivial. Can we trust ourselves to keep an eye on this?

Comment (markfickett) [reply api/libproject/devstart.rb]: OK!

**Pull request merged**